Added path_local, path_continuous_u and path_joined properties to CSGPolygon

BastiaanOlij opened the pull request:
Add three new properties to CSGPolygon.

**path_local** toggles whether the transform of the Shape is used as our starting point or the transform of the polygon node. The default behavior was the use the transform from the Shape so the extrusion follows the path.

The problem with this approach is that if you want multiple shapes to follow the same path, they can't be moved into location. 

When the new property is turned on we use the transform of the CSGPolygon and the path is in essence transposed to the polygons location. We can now use the same path for multiple extrusion such as in this bridge:

![image](https://user-images.githubusercontent.com/1945449/42217164-81cf237a-7f07-11e8-95df-a27dfcd327c1.png)

**path_continuous_u** ensures the u component of our uv keeps increasing as the mesh is extruded along the path. This allows proper tiling of textures regardless of any scale that is applied to the UV.

**path_joined** will leave out the "end caps" and instead join the start of our shape to the end of our shape. 
While you can make sure that your curve nicely ends at the same position as it starts, getting the angle at which they join is near impossible resulting in a mismatch between start and end. 

*Bugsquad edit:* Fixes #19915.

BastiaanOlij committed:
Added path_local and path_continious_u properties to CSGPolygon

akien-mga commented on modules/csg/csg_shape.cpp:
```diff
@@ -2003,6 +2007,9 @@ void CSGPolygon::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_path_rotation"), &CSGPolygon::get_path_rotation);
 
+	ClassDB::bind_method(D_METHOD("set_path_local", "local"), &CSGPolygon::set_path_local);
+	ClassDB::bind_method(D_METHOD("get_path_local"), &CSGPolygon::get_path_local);
```
Boolean getters should start with `is_`. Also the argument for the setter would be better as `enabled` than `local`.

BastiaanOlij replied:
so is_path_local and set_path_local?

toger5 commented:
to be honest I do not completely understand the property description. 
Donno If it's just me. But it won't hurt to make it a little longer and more precise. Maybe this will help some ppl? (It Would help me :) 
)

BastiaanOlij commented:
@toger5 I'm open for suggestions, the difference is simply that the path is extruded centered on the path node if its false, and centered from the CSGPolygon node when true. Very handy when you have multiple CSGPolygon nodes that all extrude different shapes along the same path (or the same shape at different locations as in my bridge example). Without this setting all shapes would all just be in the same position making it really useless (tbh I can't really think of a use case where you want to original logic but that's just me)

BastiaanOlij commented:
Ok, changed method names as suggested by Akien and added one more extra property called path_continuous_u. This sets the U component of the UVs to continiously increase based on the distance traveled along the path. Great for applying tilable road textures to a road :)
